src/components/atoms/SpacesDropdown/SpacesDropdown.tsx

```diff
@@ -69,7 +67,9 @@ export const SpacesDropdown: React.FC<SpacesDropdownProps> = ({
       return "Select a space";
     }
 
-    const space = venueSpaces.find(({ name }) => name === spaceValue);
+    const space =
+      venueSpaces.find(({ name }) => name === spaceValue.name) ?? defaultSpace;
```

Comment:
This can be made faster (`O(n) -> O(1)` )  if `venueSpaces` or whatever its new name isn't an array, but an object of type `Record` with the key being the id of the space. Display doesn't require the number of fields in `SpacesDropdownPortal ` to be limited to name and template only.

Reply:
redone `venueSpaces` so that it's an obj with a `Record` type 6b9f914